qlib/workflow/__init__.py

```diff
@@ -370,11 +370,11 @@ def uri_context(self, uri: Text):
             the temporal uri
         """
         prev_uri = self.exp_manager.default_uri
-        C.exp_manager["kwargs"]["uri"] = uri
```

Comment:
Hi,   @ChiahungTai  

I try to explain the design of the experiment system. It is a little complicated.  Discussions are welcome if I fail to make it clear.

The `ExpManager` is expected to be a singleton and the global Config (e.g. `C`)  is also a singleton (btw, we can have multiple `Experiment`s with different uri). 
So we try to align them together.  So you can see that  [they share the same variable](https://github.com/microsoft/qlib/blob/main/qlib/workflow/expm.py#L255), which is called **default uri**

When the user starts an experiment, the user may want to set the uri to a specific uri (it will override **default uri** during this period),  and then unset the **specific uri** and fallback to the **default uri**.    `_current_uri` is that **specific uri**.
The uri shared by `ExpManager ` and `C` is the **default uri**.

Originally, `uri_context` tries to temporarily set the `default_uri`. So the shared uri should be changed instead of  `_current_uri`.  (sometimes user doesn't start an experiment,  so there is no `_current_uri`.  but user want to query )

I think the problem is [here](https://github.com/microsoft/qlib/blob/main/qlib/workflow/expm.py#L28). The `_current_uri` should not be set and `uri` should be ignored because of the shared mechanism between two singletons.

You can upload your issue as a test case like [this](https://github.com/microsoft/qlib/blob/main/tests/test_all_pipeline.py#L74).  It will be helpful for other users.

Reply:
Please ignore my reply.
I think the current design is not good enough.  I'll try to create a new PR.  You are welcome to review my new PR later.

You can upload your issue as a test case like [this](https://github.com/microsoft/qlib/blob/main/tests/test_all_pipeline.py#L74). It will be helpful for other users.  Hope it would work right in my new PR. 

Thanks.

Reply:
@you-n-g 
I would love to review your patch. I will try to write a test case for reproducation. I guess the problem is the missing of call ExpManager constructor too.
I can also help to review qlib/data part. I have done a class of DBStorage backend to support SQLite as qlib format backend in my own project. 

BTW, I will reply #1326 after I finish a spirit in my own project. I can read/write Chinese. If something hard to explain in English, feel free to use Chinese.

Reply:
@ChiahungTai  
I'm excited about your great passion for Qlib.
I have invited you as a collaborator in Qllib.   Hope that would make the collaboration smoother
众人拾柴火焰高. haha~